scripts/python/qa-scripts/createVCAAggregate.py

```diff
@@ -0,0 +1,523 @@
+        self.dfMasterProposers = self.gspreadWrapper.dfMasterProposers.set_index('id')
+        # Set all counters to 0
+        self.dfVca[self.opt.noVCAReviewsCol] = 0
+        for col in self.allColumns:
+            self.dfVca[col] = 0
+            self.dfVca['Result ' + col] = 0
+
+    def prepareVCAsFileList(self):
+        for currentDirPath, currentSubdirs, currentFiles in os.walk('./vcas-files'):
+            for aFile in currentFiles:
+                if aFile.endswith(".csv") :
+                    fpath = str(os.path.join(currentDirPath, aFile))
+                    self.vcasFiles.append(fpath)
+
+    def loadVCAsFiles(self):
+        self.prepareBaseData()
+        self.prepareVCAsFileList()
+        self.vcasData = []
+        self.vcasFileList = []
+        for vcaFile in self.vcasFiles:
+            print("Loading {}".format(vcaFile))
+            data = pd.read_csv(vcaFile)
+            data.set_index(self.opt.assessmentsIdCol, inplace=True)
+            data.fillna('', inplace=True)
+            if not set([self.opt.notValidCol]).issubset(data.columns):
+                data[self.opt.notValidCol] = data[self.opt.notValidAlternativeCol]
+                data.drop(self.opt.notValidAlternativeCol, axis=1, inplace=True)
+            data = self.filterVCAConficts(data, vcaFile)
+            if data is not False:
+                self.vcasData.append(data)
+                self.vcasFileList.append(vcaFile)
+
+    def createDoc(self):
+        self.loadVCAsFiles()
+        # Loop over master ids as reference
+        for id, row in self.dfVca.iterrows():
+            proposerAss = self.dfMasterProposers.loc[id]
+            # Loop over all vca files
+            for filesIdx, vcaDf in enumerate(self.vcasData):
+                if (id in vcaDf.index):
+                    locAss = vcaDf.loc[id]
+                    integrity = self.checkIntegrity(id, row, locAss)
+                    vca_fn = self.vcasFileList[filesIdx]
+                    vca_filename = os.path.basename(vca_fn.replace('\\',os.sep))
+                    single_vca = next((item for item in self.vcas if item['vca_file'] == vca_filename), None)
+                    if (integrity is False):
+                        print("{} failed to pass the integrity test at id {}".format(vca_fn, id))
+                    if integrity:
+                        bad = self.badFeedback(locAss)
+                        good = self.goodFeedback(locAss)
+                        excellent = self.excellentFeedback(locAss)
+                        if (self.isVCAfeedbackValid(locAss, bad, good, excellent)):
+                            if (bad or good or excellent):
+                                reviews_num_col = "No. of Reviews"
+                                if locAss[self.opt.challengeCol] in self.dChallenges:
+                                    reviews_num_col = reviews_num_col + " " + locAss[self.opt.challengeCol]
+                                if reviews_num_col in single_vca:
```

Comment:
How can this happen? Also this part would benefit from an explaining comment

Reply:
There is the need to keep track of the reviews made by vCA for specific challenges (like Natives COTI). 
In the first part I create the base column name, that changes if the assessments is for a challenge in `dChallenges`. 
Then I update the `single_vca` dict: if it already has the key it increases the counter, otherwise it creates the key with value 1.